### ISSUE TITLE
Make it possible to disable high DPI mode

### ISSUE THREAD

**Issue by fossegutten** [17cupsofcoffee/tetra#154]
When high DPI scaling in Windows is not integer scaling ( 200%, 300% ..), the pixels get blurred (250%, 175% ..). Could be avoided with being allowed to disable scaling of the window.

Example pic taken with pixel perfect show all mode. Look at how crisp the mouse icon in windows is, compared to texture in Tetra.

![test_game_tetra_rust_R3hfdNGcsF](https://user-images.githubusercontent.com/15088582/70851650-138b9b00-1e98-11ea-82da-ed635ad90959.png)

**Comment by 17cupsofcoffee** [comment]
DPI handling is something I definitely want to take a look at once 0.3 is out (which should be in about 30 seconds if the CI passes...) - it's a definite weak point of Tetra right now.

**Comment by fossegutten** [comment]
I just wanted to say I made highdpi work in Windows 10, by adding some manifest xml file. Using the `.high_dpi(true/false)` made no difference in Windows. 

You can also hack it like this in windows:
![83754932-7440a700-a675-11ea-9546-a066868e259d](https://user-images.githubusercontent.com/15088582/104584293-a500aa00-5662-11eb-9602-12438b64f20c.png)

**Comment by 17cupsofcoffee** [comment]
Ah, cool :+1: Things work similar on Mac, I think - to get `.high_dpi(true)` to do anything pre-Catalina you have to set a flag in the `.app` package.

If you can send me any details of the manifest changes you made to enable this, I can add them to the docs somewhere?

**Comment by fossegutten** [comment]
Here is how I did it! Found it piece by piece, by googling around.

1. add this to Cargo.toml

```
[build-dependencies]
embed-resource = "1.4"
```
2. Create a `build.rs` file in main folder of repository, with this code:
```
extern crate embed_resource;

fn main() {
    embed_resource::compile("manifests.rc");
}
```
3. create the `manifests.rc` file that you mentioned in `build.rs` file, with this code:
```
#define RT_MANIFEST 24
1 RT_MANIFEST "manifest.xml"
```
4. Add the `manifest.xml` file, i used this code:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0" xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
  <asmv3:application>
    <asmv3:windowsSettings>
      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2</dpiAwareness>
    </asmv3:windowsSettings>
  </asmv3:application>
</assembly>
```

**Comment by 17cupsofcoffee** [comment]
Cool, thank you! I will add some more info to the docs/website before I close this issue off. [They do currently mention that you need a manifest](https://docs.rs/tetra/0.5.6/tetra/struct.ContextBuilder.html#method.high_dpi), but they don't mention exactly *which* settings you need, or how to embed them via Rust, so I think there's room for improvement there.

**Comment by 17cupsofcoffee** [comment]
New docs:

![image](https://user-images.githubusercontent.com/784533/104632105-207b4f00-5695-11eb-847a-9f7e22cfada2.png)

I did read that `true/pm` is slightly preferable to `true` for `dpiAware`, as this gives you per-monitor DPI awareness on older versions of Windows that don't support `permonitorv2`.

Let me know if you think this needs any tweaks, otherwise I'll (finally!!!) close this issue :smile:

**Comment by 17cupsofcoffee** [comment]
I published the docs changes in 0.5.7, and added an entry about high-DPI to the FAQ on the website. Going to close this issue for now, but feel free to reopen if you think any more tweaks need making. Thank you for all your help and patience with this issue :)
